modules/javafx.web/src/main/native/Source/WebCore/platform/java/FileSystemJava.cpp

```diff
@@ -232,21 +232,75 @@ String openTemporaryFile(const String&, PlatformFileHandle& handle)
     return String();
 }
 
-PlatformFileHandle openFile(const String&, FileOpenMode)
+PlatformFileHandle openFile(const String& path, FileOpenMode mode)
 {
-    notImplemented();
-    return invalidPlatformFileHandle;
+    if (mode != FileOpenMode::Read) {
+        return invalidPlatformFileHandle;
+    }
+
+    JNIEnv* env = WebCore_GetJavaEnv();
+    static jmethodID mid = env->GetStaticMethodID(
+            GetFileSystemClass(env),
+            "fwkOpenFile",
+            "(Ljava/lang/String;)Ljava/io/InputStream;");
+    ASSERT(mid);
+
+    PlatformFileHandle result = env->CallStaticObjectMethod(
+            GetFileSystemClass(env),
+            mid,
+            (jstring)path.toJavaString(env));
+
+    CheckAndClearException(env);
+    return result ? result : invalidPlatformFileHandle;
 }
 
-void closeFile(PlatformFileHandle&)
+void closeFile(PlatformFileHandle& handle)
 {
-    notImplemented();
+    if (isHandleValid(handle)) {
+        JNIEnv* env = WebCore_GetJavaEnv();
+        static jmethodID mid = env->GetStaticMethodID(
+                GetFileSystemClass(env),
+                "fwkCloseFile",
+                "(Ljava/io/InputStream;)V");
+        ASSERT(mid);
+
+        env->CallStaticVoidMethod(
+                GetFileSystemClass(env),
+                mid, handle);
+        CheckAndClearException(env);
+        handle = invalidPlatformFileHandle;
+    }
 }
 
-int readFromFile(PlatformFileHandle, char*, int)
+int readFromFile(PlatformFileHandle handle, char* data, int length)
 {
-    notImplemented();
-    return -1;
+    if (length < 0) {
+        return -1;
+    }
+
+    JNIEnv* env = WebCore_GetJavaEnv();
+
```

Comment:
I see blank line after each line of code. Probably you can group the statements accordingly.